src/utils/colonyClient.ts

```diff
@@ -20,8 +27,31 @@ export const getCachedColonyClient = async (
     colonyClientsCache[colonyAddress] = colonyClient;
     return colonyClient;
   } catch (error) {
+    console.error(
+      `Unable to fetch colony client for address: ${colonyAddress}`,
+      error,
+    );
+  }
+
+  return;
+};
 
+export const getColonyFromDB = async (
+  colonyAddress: string,
+): Promise<Colony | undefined> => {
+  const { data } =
+    (await query<GetColonyQuery, GetColonyQueryVariables>(GetColonyDocument, {
+      id: colonyAddress,
+    })) ?? {};
+
+  const colony = data?.getColony;
+
+  if (!colony) {
+    console.error(
+      `Could not find colony: ${colonyAddress} in database. This is a bug and should be investigated.`,
```

Comment:
I feel like this log might become irrelevant when this util function is reused somewhere else (and passes a wrong colony address)